### ISSUE TITLE
[BUG] Tabular models fail to function properly in certain Docker container environments.

### ISSUE THREAD

**Issue by Anyon123** [autogluon/autogluon#4895]
**Bug Report Checklist**

- [ ] I provided code that demonstrates a minimal reproducible example.  
- [√] I confirmed bug exists on the latest mainline of AutoGluon via source install.  
- [√] I confirmed bug exists on the latest stable version of AutoGluon.  

**Describe the bug**

When running tabular models such as DirectTabular or RecursiveTabular in a Docker container where the number of logical CPUs is less than the number of physical CPUs, a bug occurs.

It appears that every tabular model class has overridden a method named _get_default_resources(). For instance, in line 518 of tabular/models/lgb/lgb_model.py, there is an annotation stating that logical=False is faster for training. However, in practice, setting logical=False can lead to bugs in certain Docker environments.

Given this issue, why not change the setting to logical=True, or simply remove this overridden method altogether, since the parent class already implements this method? This adjustment could help avoid the bug and ensure more consistent behavior across different environments.

**Expected behavior**

**To Reproduce**

**Screenshots / Logs**

**Installed Versions**

<details>

```python
# Replace this code with the output of the following:
from autogluon.core.utils import show_versions
show_versions()
```

</details>

**Comment by Innixma** [comment]
Hello, thanks for submitting an issue!

Please provide a code example + stack trace from the exception, as well as logs of the training.

It is currently unclear what the problem is, as the bug mentioned isn't described. What is going wrong?

Additionally, how are the number of logical CPUs less than the number of physical CPUs? I have not experienced this before.

**Comment by shchur** [comment]
Fixing the issue on the tabular side should automatically resolve it for the TS models, so I will remove the TS tag for now.

**Comment by Anyon123** [comment]
Thank you for the previous responses, but it seems this issue still exists in version 1.6.1.

My Environment:
- AutoGluon version: 1.6.1
- Container runtime: Docker / Kubernetes
- Host physical cores: 64
- Container CPU quota: 20

Problem

  When running AutoGluon Tabular tasks in a containerized environment, the assertion at line 1146 assert system_num_cpus >= num_cpus in AbstractModel._calculate_total_resources fails.

  Root Cause

  system_num_cpus and default_num_cpus use different CPU counting strategies:

   - Line 1043 system_num_cpus = resource_manager.get_cpu_count() → calls joblib.cpu_count(only_physical_cores=False), which respects the container CPU quota, returning 20
   - Line 1049 default_num_cpus = self._get_default_resources() → For models like XGBoost, LightGBM, CatBoost, etc., default_resources_physical_cores_only = True, which calls
     joblib.cpu_count(only_physical_cores=True) → internally uses psutil.cpu_count(logical=False), which reads the host machine's physical core count, returning 64

  Later in the logic, num_cpus is assigned default_num_cpus=64, causing the assertion 20 >= 64 to fail.

  The fundamental issue is that psutil.cpu_count(logical=False) is not aware of cgroup CPU quotas in containers and returns the host machine's physical core count instead.

  Reproduction Conditions

   1. Running in a container (Docker/K8s) where the CPU quota is less than the host machine's physical core count
   2. Using an AutoGluon Tabular model with default_resources_physical_cores_only = True (e.g., XGBoost, LightGBM, CatBoost, FastAI, TabNet, etc.)
   3. Not explicitly specifying num_cpus, letting it fall through the "auto" logic path

  Reproduction Steps

   1 from autogluon.tabular import TabularPredictor
   2 
   3 # Run in a container where CPU quota < host physical cores
   4 predictor = TabularPredictor(label='target').fit(train_data)
   5 # Assertion fails: assert system_num_cpus >= num_cpus  (20 >= 64)


Here is Suggested Fix from AI, FYI:
Cap `default_num_cpus` and `default_num_gpus` at the system's actually available resources, right after `_get_default_resources()` is called in `_calculate_total_resources`:
  default_num_cpus, default_num_gpus = self._get_default_resources()
  default_num_cpus = min(default_num_cpus, system_num_cpus)
  default_num_gpus = min(default_num_gpus, system_num_gpus)
